#Moving Obstacle #3

#Number of orders:
#Number of shipments per order: 1
#Challenges:
#  -Interruption: Yes
#  -Drop parts: No
#  -Sensor blackout: No
#  -Part flipping: Yes
#  -Faulty parts: Yes
#  -Moving obstacle: Yes
#Containers
# -Belt: Yes
# -Bins: Optional
# -Shelf: Yes
#AGV
# -order_0:
#   -shipment_0: agv1
# -order_1:
#   -shipment_0: agv2
#Time limit: Yes

time_limit: 500

options:
  gazebo_state_logging: true
  insert_models_over_bins: true
  insert_models_over_shelves: true
  belt_population_cycles: 8
  spawn_extra_models: false
  model_type_aliases:
    order_part1: piston_rod_part_blue
    order_part2: gasket_part_red
    order_part3: gasket_part_green
    order_part3: pulley_part_red

shelf_layout:
  row_1: ['base',0,'base','base'] #shelves: 3, 4, 5
  row_2: ['pipe','base',0,'base']   #shelves: 6, 7, 8
  row_3: ['base',0,'collar','collar'] #shelves: 9, 10, 11

aisle_layout:
  person_1:
    location: -1.57
    start_time: 0.
    move_time: 9.
    wait_time: 7.
  #Aisle 3
  person_2:
    location: 1.57
    start_time: 16.
    move_time: 9.
    wait_time: 7.


orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 1
    destinations: [agv1]
    products:
      part_0:
        type: pulley_part_red
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: ['pi', 0, 0]
      part_1:
        type: piston_rod_part_blue
        pose:
          xyz: [0.15, 0.2, 0]
          rpy: [0, 0, '-pi/4']
      part_2:
        type: gasket_part_red
        pose:
          xyz: [-0.15, -0.2, 0]
          rpy: [0, 0, 'pi/4']
  order_1:
    announcement_condition: wanted_products
    announcement_condition_value: 1
    shipment_count: 1
    destinations: [agv2]
    products:
      part_0:
        type: pulley_part_red
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: ['pi', 0, 0]
      part_1:
        type: gasket_part_green
        pose:
          xyz: [-0.15, -0.2, 0]
          rpy: [0, 0, '-pi/4']


belt_models:
  piston_rod_part_blue:
    15.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 'pi/2']
models_over_bins:
  bin8:
    models:
      gasket_part_red:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 3
        num_models_y: 3


models_over_shelves:
  shelf8:
    models:
      gasket_part_green:
        xyz_start: [1, -0.3, 0.0]
        xyz_end: [0., -0.4, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 8
        num_models_y: 1
      pulley_part_red:
        xyz_start: [1, 0.45, 0.0]
        xyz_end: [-0.5, 0.45, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 4
        num_models_y: 1
    shelf1:
      models:
        gasket_part_red:
          xyz_start: [1, -0.3, 0.0]
          xyz_end: [0., -0.4, 0.0]
          rpy: [0, 0, 'pi/4']
          num_models_x: 8
          num_models_y: 1

#Faulty products
faulty_products:
  - piston_rod_part_blue_1
  - gasket_part_red_1
